CIP-0071/README.md

```diff
@@ -0,0 +1,268 @@
+
+1. A ballot box that can be redeemed at any time by a tx signed by the authorized vote counter
+2. A ballot box that can be redeemed only after polls close
+3. A ballot box that can be redeemed once a majority of voters have sent in a ballot
+4. A ballot box that can be redeemed only by the specific wallet specified in the `voter` datum of each UTxO
+5. A ballot box that can be redeemed only after polls close, has to burn the ballots it redeems, and has to send the minUTxO back to the voter address
+
+Because the ballot creation and vote casting process has already occurred on-chain we want to provide the maximum flexibility in the protocol here so that each project can decide what is best for their own community.  Helios code for the simple case enumerated as #1 above would be:
+
+```ts
+const EXPECTED_SIGNER: PubKeyHash = PubKeyHash::new(#0123456789abcdef)
+
+func main(ctx: ScriptContext) -> Bool {
+  ctx.tx.is_signed_by(EXPECTED_SIGNER)
+}
+```
+
+### The Vote Counting Process
+
+#### "Ballot Counter" -> Authorized Wallet
+
+Given the flexible nature of the ["ballot box" smart contract](#Ballot-Box---Smart-Contract) enumerated above, we propose a simple algorithm for counting votes and returning the ballots to the user:
+
+1. Ensure the polls are closed (can be either on or off-chain)
+2. Iterate through all UTxOs in forward-time-order locked in the "ballot box" and for each:
+	* Determine which assets are inside of that UTxO
+	* Mark their most recent vote to match the `vote` object in the UTxOs datum
+3. Ensure any required quorums or thresholds were reached
+4. Report on the final ballot outcome
+
+Javascript-like pseudocode using the [Lucid library](https://github.com/spacebudz/lucid) for the above algorithm would be as follows:
+
+```js
+function countVotes(ballotPolicyId, ballotBox) {
+  var votesByAsset = {};
+  const votes = await lucid.utxosAt(ballotBox);
+  for (const vote of votes) {
+    const voteResult = Data.toJson(Data.from(vote.datum));
+    for (const unit in vote.assets) {
+      if (!unit.startsWith(ballotPolicyId)) {
+        continue;
+      }
+      const voteCount = Number(vote.assets[unit]);         // Should always be 1
+      votesByAsset[unit] = {
+        voter: voteResult.voter,
+        vote: voteResult.vote,
+        count: voteCount
+      }
+    }
+  }
+  return votesByAsset;
+}
+```
+
+There is no requirement that the "ballot counter" redeem all "ballots" from the "ballot box" and send them back to the respective voters, but we anticipate that this is what will happen in practice.  We encourage further open-sourced code versions that enforce this requirement at the smart contract level.
+
+### Reclaiming ADA Locked by the Ballot NFTs
+
+Even if the ballot NFT is returned to the user, this will leave users with ADA locked alongside these newly created assets, which can impose a financial hardship for certain project users.
```

Comment:
```suggestion
Even if the ballot NFT is returned to the user, this will leave users with ada locked alongside these newly created assets, which can impose a financial hardship for certain project users.
```